src/system-health/health_checker/manager.py

```diff
@@ -0,0 +1,101 @@
+class HealthCheckerManager(object):
+    """
+    Manage all system health checkers and system health configuration.
+    """
+    STATE_BOOTING = 'booting'
+    STATE_RUNNING = 'running'
+    boot_timeout = None
+
+    def __init__(self):
+        self._checkers = []
+        self._state = self.STATE_BOOTING
+
+        from .config import Config
+        self.config = Config()
+        self.initialize()
+
+    def initialize(self):
+        """
+        Initialize the manager. Create service checker and hardware checker by default.
+        :return: 
+        """
+        from .service_checker import ServiceChecker
+        from .hardware_checker import HardwareChecker
+        self._checkers.append(ServiceChecker())
+        self._checkers.append(HardwareChecker())
+
+    def check(self, chassis):
+        """
+        Load new configuration if any and perform the system health check for all existing checkers. 
+        :param chassis: A chassis object.
+        :return: A tuple. The first element indicate the status of the checker; the second element is a dictionary that
+        contains the status for all objects that was checked.
+        """
+        from .health_checker import HealthChecker
+        HealthChecker.summary = HealthChecker.STATUS_OK
+        stats = {}
+        self.config.load_config()
+        # check state first to avoid user change boot timeout in configuration file
+        # after finishing system boot
+        if self._state == self.STATE_BOOTING and self._is_system_booting():
+            self._set_system_led(chassis, self.config, 'booting')

```

Comment:
1. _can we make sure if the default configuration works for all other platforms?_
It depends on how we define "works". Like you said, there might be platforms doesn't support setting system led or lack of some information in DB, per the default configuration, it would either trigger a warning syslog or an error message in the output of "show system-health summary" command. 
2. _The default boot up time is 300s which is 5mins. Until uptime passes 5 mins, does it stay in booting state even for the case that all the service and daemon are already running?_
Yes. Now system monitor service is depending on monit service to determine the status of critical services/process/file systems. monit service has a configuration file which set the start delay to 300s for itself. System monitor service will always try reading that config file to determine the wait time. If monit configuration file doesn't exist, it will use default 300s.
3. _why this system health monitoring service needs to set the system health led?_
This is according to the design. The system led is a physical component that using for indicating the status of the switch, so if there is anything wrong in the system, the led status should be red to inform user that something bad is happending.
4. _It will be safe to disable this health monitoring service unless the platform specific configuration file exists_
Yes, it is safer to do that. For now, only mellanox platform has such configuration file. @jleveque what's your opinion about this?

Reply:
Added a check to system health configuration file, if the file does not exist, healthd just exit.

Reply:
I think it is fine for now to log a message and exit if the config file doesn't exist, as you're now doing.